### PR TITLE
Include `db.system` attribute in `opentelemetry_ecto`

### PR DESCRIPTION
**Pull request by tomtaylor**
Added clauses for the major Ecto adapters, and refactored how attributes are assembled, to make it easier to read.

### CODE REVIEW

**Review by tsloughter — APPROVED** [review]
Looks good to me. @bryannaegele any thoughts?

File: instrumentation/opentelemetry_ecto/lib/opentelemetry_ecto.ex
Comment (tsloughter) [inline instrumentation/opentelemetry_ecto/lib/opentelemetry_ecto.ex:99]: This isn't guaranteed, right? A Repo could be any storage?

Comment (tomtaylor) [reply instrumentation/opentelemetry_ecto/lib/opentelemetry_ecto.ex:99]: Yes, true. Maybe better falling back to `nil` then?

Comment (andrewhr) [reply instrumentation/opentelemetry_ecto/lib/opentelemetry_ecto.ex:99]: What about default to `:other_ecto`? This should avoid a `nil` on a required attribute.

Comment (tomtaylor) [reply instrumentation/opentelemetry_ecto/lib/opentelemetry_ecto.ex:99]: Yeah, makes sense to me. And if there’s no value we should just no send the property at all, right? Rather than a `nil`?

Comment (andrewhr) [reply instrumentation/opentelemetry_ecto/lib/opentelemetry_ecto.ex:99]: The `ecto` (or `other_ecto`) instead of nil, given it's a required field is too much a stretch?  What do you think, @tsloughter?

Comment (tomtaylor) [reply instrumentation/opentelemetry_ecto/lib/opentelemetry_ecto.ex:99]: Shall we just drop this attribute if it doesn't match right now? For 99% of users that gets this attribute in and working, and we can revisit the fallback in the future?

Comment (tsloughter) [reply instrumentation/opentelemetry_ecto/lib/opentelemetry_ecto.ex:99]: Yea, probably the best way forward.

**Review by andrewhr — APPROVED** [review]

**Comment by andrewhr** [comment]
Can't wait for it to get merged @tomtaylor, great work! 🚀

**Review by tsloughter — APPROVED** [review]

**Comment by tsloughter** [comment]
Will probably pass once I merge 188 and then will merge this.

**Comment by tsloughter** [comment]
Will merge after the conflicts are resolved.

**Comment by tsloughter** [comment]
@tomtaylor ping

**Comment by tomtaylor** [comment]
This should be ready to go now.

**Comment by tsloughter** [comment]
Thanks!

**Pull request merged**
